### PR TITLE
[processing] use hours as cost units for service area algorithms (fix #30464)

### LINKED ISSUE #30464: Homogenize travel cost units within network analysis algorithms
When running shortest path algorithm (eg, point to point) with the "fastest" type of travel, the cost field in the output layer is in "hours".
When running the service area algorithms with the "fastest" travel cost, "seconds" are expected as input. Since the latter is not obviously mentioned, this may lead to errors.

It could be nice we use the same units across those algorithms and explicitly expose the units in use.

### PR DESCRIPTION
**Pull request by alexbruy**
## Description
Use hours as units for "travel cost" parameter in the service area algorithms (when "fastest" strategy is used). This makes them consistent with the output of shortest path algorithms. Fixes #30464.

### CODE REVIEW

**Comment by nyalldawson** [comment]
This could potentially break existing models, couldn't it?

**Comment by alexbruy** [comment]
The only breakega is that default value (if exists) for cost parameter should be updated.

**Comment by nyalldawson** [comment]
> The only breakage is that default value (if exists) for cost parameter should be updated.

That's what I'm concerned about. What do you think about adding a new parameter for default in hours, and hiding the old one and only using it when the new one isn't specified?

**Comment by alexbruy** [comment]
Probably then better to leave it as is to avoid another breakage in future when old hidden parameter will be removed and hours parameter will be renamed to the old consistent name?

**Comment by alexbruy** [comment]
Ok, I have added new parameted and hide the old one. Still not sure if this is a right way, as we will need to remove and rename duplicated parameters at some stage.

File: src/analysis/processing/qgsalgorithmserviceareafromlayer.cpp [hunk 1]
```diff
@@ -78,6 +87,12 @@ QVariantMap QgsServiceAreaFromLayerAlgorithm::processAlgorithm( const QVariantMa
     throw QgsProcessingException( invalidSourceError( parameters, QStringLiteral( "START_POINTS" ) ) );
 
   double travelCost = parameterAsDouble( parameters, QStringLiteral( "TRAVEL_COST" ), context );
+  if ( !travelCost )
+    travelCost = parameterAsDouble( parameters, QStringLiteral( "TRAVEL_COST2" ), context );
```
Comment (nyalldawson) [inline src/analysis/processing/qgsalgorithmserviceareafromlayer.cpp:91]:
```suggestion
  // use older deprecated travel cost style if specified, to maintain old api
  const bool useOldTravelCost = parameters.value( QStringLiteral("TRAVEL_COST" ) ).isValid();
  double travelCost = parameterAsDouble( parameters, useOldTravelCost ? QStringLiteral( "TRAVEL_COST" ) : QStringLiteral( "TRAVEL_COST2" ), context );
```

File: src/analysis/processing/qgsalgorithmserviceareafromlayer.cpp [hunk 2]
```diff
@@ -78,6 +87,12 @@ QVariantMap QgsServiceAreaFromLayerAlgorithm::processAlgorithm( const QVariantMa
...
+
+  int strategy = parameterAsInt( parameters, QStringLiteral( "STRATEGY" ), context );
+  if ( strategy )
```
Comment (nyalldawson) [inline src/analysis/processing/qgsalgorithmserviceareafromlayer.cpp:94]:
```suggestion
  if ( strategy && !useOldTravelCost )
```

File: src/analysis/processing/qgsalgorithmserviceareafromlayer.cpp (hunk 1 above)
Comment (nyalldawson) [reply src/analysis/processing/qgsalgorithmserviceareafromlayer.cpp:91]: Same changes apply to qgsalgorithmserviceareafrompoint.cpp

**Review by nyalldawson — APPROVED** [review]

**Comment by alexbruy** [comment]
Don't know why Processing test fails of Travis, it passes locally.

**Pull request merged**